src/ngraph/runtime/gpu/gpu_emitter.cpp

```diff
@@ -516,6 +516,36 @@ namespace ngraph
                     return;
                 }
                 auto reshape = static_cast<const op::Reshape*>(node);
+                auto can_skip_reshape = [&]() {
+                    if (!reshape->get_is_transpose())
```

Comment:
Please don't do this! Emitter isn't a MemoryLayout pass. The decision whether to copy or not should be as localized as possible.
Do this:
```cpp
if (args[0].get_name() == out[0].get_name())
{
   // print skipping comment
}
else
{
  // emit a copy
}
```

Reply:
@Krovatkin 
anything to do with `op_annotations->add_in_place_oi_pair({0, 0, false});`? should we use true here?

Reply:
Logical `Reshape`s are non-destructive, so it should be `false`. It's `true` for `ReplaceSlice`, for example. The reason why CPU has those checks in Emitter is somewhat convoluted and it has to do with a `CPULayout` pass optimizing away unnecessary transposes. You guys shouldn't have these issues so a simple equal comparison should work.